src/iris/api.py

```diff
@@ -1256,6 +1261,13 @@ def on_post(self, req, resp):
         }
 
         dynamic_indices = set()
+
+        # for backwards copatibility check if optional is not defined and set it to 0 if it isn't
```

Comment:
Instead of iterating through twice, modify the step right before executing the query.

Reply:
Line 1286 is a good place (`step['step'] = index`)

Reply:
Changed.